src/geohash.h

```diff
@@ -90,6 +90,25 @@ typedef struct {
     GeoHashBits south_west;
 } GeoHashNeighbors;
 
+#define CIRCULAR_TYPE 1
+#define RECTANGLE_TYPE 2
+
+typedef struct {
+    int type;
+    double xy[2];
+    double conversion;
+    union {
+        // CIRCULE
+        double radius;
+        // RECTANGLE
+        struct {
+            double height;
+            double width;
+            double bounds[4];
+        } r;
```

Comment:
why do we have both width and height and also bounds?
maybe i'm missing something, in which case it would better be described in a comment.

Reply:
considering that `bounds` is a cache, we can either completely eliminate it (whoever needs it can re-compute it on the spot, so that the GeoShape struct only contains the needed data to specify the shape.
i.e. having a cache is a redundant (duplicate) info that is hard to tell if updated or not.

or alternatively at least move it to serve the radius too.
but also, i don't like the fact it's an array of 4 elements (hard to tell which ones is x and which ones are y), so more comments are needed to document its purpose and meaning.